internal/topo/planner/ProjectSetPlan.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 EMQ Technologies Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package planner
+
+type ProjectSetPlan struct {
```

Comment:
I am not sure if it is necessary to have project & projectSet together in one plan. Can you find the benefits when doing implementation?

Reply:
We will use `ProjectOp` for `ProjectSet` Executor. Thus we directly build a built-in `ProjectPlan` for `ProjectSetPlan` in order to build `ProjectOp` by `ProjectPlan`.

On the other hand, `ProjectSetPlan` do quite the similar thing as `ProjectPlan`, except `ProjectSet` will handle the `srf`

Reply:
I have no idea now. Just think about if it is better to keep each plan "single responsibility". We can pipe from SRF to project.

Reply:
I got it. So the final plan will be `ProjectSet` -> `Project` -> .... , that's ok to me.

Reply:
Just an alternative way. During implementation, you may have more ideas about which way is better.